pkg/cmd/registry/artifacts/crud/delete/delete.go

```diff
@@ -0,0 +1,183 @@
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/factory"
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/flag"
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/registry/artifacts/util"
+	"github.com/spf13/cobra"
+)
+
+type Options struct {
+	artifact string
+	group    string
+
+	registryID string
+
+	outputFormat string
+	force        bool
+
+	IO         *iostreams.IOStreams
+	Config     config.IConfig
+	Connection factory.ConnectionFunc
+	Logger     func() (logging.Logger, error)
+	localizer  localize.Localizer
+}
+
+func NewDeleteCommand(f *factory.Factory) *cobra.Command {
+	opts := &Options{
+		Config:     f.Config,
+		Connection: f.Connection,
+		Logger:     f.Logger,
+		IO:         f.IOStreams,
+		localizer:  f.Localizer,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Deletes all of the artifacts that exist in a given group.",
+		Long: `
+Deletes all of the artifacts that exist in a given group. 
+
+Delete command works in two modes:
+
+	- When --artifact argument is missing delete will delete all artifacts in the group
+	- When --artifact is specified delete deletes only single artifact and its version
+
+When --group parameter is missing the command will create a new artifact under the "default" group.
+		`,
+		Example: `
+## Delete all artifacts in the group "default"
+rhoas service-registry artifact delete 
+
+## Delete artifact in the group "default" with name "my-artifact"
+rhoas service-registry artifact delete artifact=my-artifact
+		`,
+		Args: cobra.RangeArgs(0, 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !opts.IO.CanPrompt() && !opts.force {
+				return flag.RequiredWhenNonInteractiveError("yes")
+			}
+
+			validOutputFormats := flagutil.ValidOutputFormats
+			if opts.outputFormat != "" && !flagutil.IsValidInput(opts.outputFormat, validOutputFormats...) {
+				return flag.InvalidValueError("output", opts.outputFormat, validOutputFormats...)
+			}
+
+			if len(args) > 0 {
+				opts.artifact = args[0]
+			}
+
+			if opts.registryID != "" {
+				return runDelete(opts)
+			}
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			if !cfg.HasServiceRegistry() {
+				return fmt.Errorf("No service Registry selected. Use 'rhoas service-registry use' use to select your registry")
+			}
+
+			opts.registryID = fmt.Sprint(cfg.Services.ServiceRegistry.InstanceID)
+			return runDelete(opts)
+		},
+	}
+
+	cmd.Flags().BoolVarP(&opts.force, "yes", "y", false, "Delete without prompt")
+	cmd.Flags().StringVarP(&opts.outputFormat, "output", "o", "json", opts.localizer.MustLocalize("registry.cmd.flag.output.description"))
+
+	cmd.Flags().StringVarP(&opts.artifact, "artifact", "a", "", "Id of the artifact")
+	cmd.Flags().StringVarP(&opts.group, "group", "g", "", "Group of the artifact")
+	cmd.Flags().StringVarP(&opts.registryID, "registryId", "", "", "Id of the registry to be used. By default uses currently selected registry")
+
```

Comment:
Should we add static completion for output flag here?

Reply:
Applying